tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -1025,22 +1072,44 @@ public void expectCompletion(long timeoutMillis, String errorMsg) throws Interru
       } // else, ok
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * @deprecated Deprecated in favor of {@link #expectError(Class, long, long, String)}.
+     */
+    @Deprecated
     public <E extends Throwable> E expectError(Class<E> clazz, long timeoutMillis, String errorMsg) throws Exception {
-      Thread.sleep(timeoutMillis);
-
-      if (env.asyncErrors.isEmpty()) {
-        return env.flopAndFail(String.format("%s within %d ms", errorMsg, timeoutMillis));
-      } else {
-        // ok, there was an expected error
-        Throwable thrown = env.asyncErrors.remove(0);
+      return expectError(clazz, timeoutMillis, timeoutMillis, errorMsg);
+    }
 
-        if (clazz.isInstance(thrown)) {
-          return (E) thrown;
+    @SuppressWarnings("unchecked")
+    final <E extends Throwable> E expectError(Class<E> clazz, final long totalTimeoutMillis,
+                                              long pollTimeoutMillis,
+                                              String errorMsg) throws Exception {
+      long totalTimeoutRemainingMillis = totalTimeoutMillis;
+      long timeStampA = System.nanoTime();
```

Comment:
As Viktor noted, there's an issue with the units here; so either Thread.sleep and `ms` (`System.currentTimeMillis()`) or sticking to nanos all the way. To be honest `ms` are likely good enough here, WDYT?

Reply:
yes, the unit mismatch will be fixed. ms sleep duration is sufficient, however time elapsed should use `nanoTime()` (due to monotonic properties of the API).